### PR TITLE
Mango python

### PR DESCRIPTION
**Pull request by akmorrow13**
TODOs:
- [x] Change naming from mango to bdg_mango after PR https://github.com/bigdatagenomics/adam/pull/1705 is merged

- [ ] Get pip for ADAM up to be referenced by Mango

### CODE REVIEW

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

File: bin/mango-notebook [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# limitations under the License.
+#
+
+set -ex
+
+# Figure out where Mango is installed
+SCRIPT_DIR="$(cd `dirname $0`/..; pwd)"
+
+# Find Mango cli assembly jar
+MANGO_CLI_JAR=
```
Comment (Georgehe4) [inline bin/mango-notebook]: Unneeded here?

File: bin/pymango [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# limitations under the License.
+#
+
+set -ex
+
+# Figure out where Mango is installed
+SCRIPT_DIR="$(cd `dirname $0`/..; pwd)"
+
+# Find Mango cli assembly jar
+MANGO_CLI_JAR=
```
Comment (Georgehe4) [inline bin/pymango]: Also here?

File: bin/mango-notebook [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+fi
+if [ -z "$PYSPARK" ]; then
+  echo "SPARK_HOME not set and spark-shell not on PATH; Aborting."
+  exit 1
+fi
+
+export PYSPARK_DRIVER_PYTHON=jupyter
+export PYSPARK_DRIVER_PYTHON_OPTS='notebook'
+
+${PYSPARK} --jars ${MANGO_CLI_JAR} --driver-class-path ${MANGO_CLI_JAR}
```
Comment (Georgehe4) [inline bin/mango-notebook]: Newline

File: mango-python/src/mango/QC.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        # make sure coverageRDD is flattened
+        flattened = coverageRDD.flatten()
+
+        # compute coverage distribution
+        coverageDistribution = flattened.toDF().rdd \
+            .map(lambda r: (r["count"], 1)) \
+            .reduceByKey(lambda x,y: x + y) \
+            .collect()
+
+        coverageDistribution.sort()
```
Comment (Georgehe4) [inline mango-python/src/mango/QC.py]: Can this operation be done in the rdd instead of the collected object?

rdd...
.sortByKey()
.collect()?

**Comment by akmorrow13** [comment]
TODOs:
- [ ] Change naming from mango to bdg_mango after PR https://github.com/bigdatagenomics/adam/pull/1705 is merged

- [ ] Get pip for ADAM up to be referenced by Mango

**Comment by fnothaft** [comment]
Hi @akmorrow13! I was meaning to chat with you about this today, but I think that https://github.com/bigdatagenomics/adam/pull/1706 is a preferable route vs https://github.com/bigdatagenomics/adam/pull/1705 for package naming. Before we move forward on the packaging here, what are your thoughts?

**Comment by akmorrow13** [comment]
@fnothaft yes https://github.com/bigdatagenomics/adam/pull/1706 seems like a much cleaner solution. I must have missed that :P

**Comment by fnothaft** [comment]
No sweat!

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Pull request merged**